model/gen-ai/registry.yaml

```diff
@@ -192,6 +192,37 @@ groups:
         type: string
         brief: Human-readable name of the GenAI agent provided by the application.
         examples: ["Math Tutor", "Fiction Writer"]
+      - id: gen_ai.agent.version
+        stability: development
+        type: string
+        brief: Human-readable version of the GenAI agent provided by the application.
+        note: >
+          AI Agent Version refers to a specific iteration of an AI-powered agent, incorporating
+          updates, improvements, and optimizations in functionality, performance, and learning
+          capabilities for enhanced decision-making and task execution.
+        examples: ["1.0"]
+      - id: gen_ai.agent.role
+        stability: development
```

Comment:
are there some existing agent frameworks or services that support it? How would it be different from the name and/or description?

Reply:
Is it used by other frameworks? We can add have crewai-specific conventions, but I'm not sure if it's used by anyone else, so it might not be the right candidate for the general attribute.

Also, would you classify it as something essential for agent observability? I mean humans can figure out the role from name and description and not sure what visualization/tools would do with role information. So is there some specific scenario where it'd be important to have the role formalized?

Reply:
MetaGPT also have roles concept https://github.com/geekan/MetaGPT/tree/main/metagpt/roles , like architect, assistant, engineer, researcher etc. But seems MetaGPT is using some other way to describe roles, let me do more research for this, thanks @lmolkova